public/app/features/alerting/unified/components/rule-editor/LabelsField.tsx

```diff
@@ -23,94 +113,173 @@ const LabelsField: FC<Props> = ({ className }) => {
   } = useFormContext<RuleFormValues>();
 
   const labels = watch('labels');
-  const { fields, append, remove } = useFieldArray({ control, name: 'labels' });
+  const { fields, remove, append } = useFieldArray({ control, name: 'labels' });
 
-  const getDistinctDropdownOptions = (
-    fields: Array<FieldArrayWithId<RuleFormValues, 'labels', 'id'>>,
-    type: AlertLabelDropdownProps['type']
-  ) =>
-    [...new Set(fields.map((field) => field[type]))]
-      .filter((field) => Boolean(field))
-      .map((field) => ({ label: field, value: field }));
+  const { loading, labelsByKey } = useGetCustomLabels(dataSourceName || RuleFormType.grafana);
```

Comment:
It seems to me we shouldn't fall back to `RuleTypeForm.grafana`. 
We should always require passing `dataSourceName` instead. `grafana` fallback might give us incorrect suggestions (creating/editing Mimir/Loki rules `grafana` suggestions are useless)

Reply:
You're right, I removed this fallback value. 
I left this prop as optional in the `LabelsField` component as based on this it either shows the dropdowns or the text inputs. The `LabelsWithSuggestions` component does make it mandatory now though. See https://github.com/grafana/grafana/pull/57783#discussion_r1015865847
Let me know what you think though :)